test/functional/inspec_vendor_test.rb

```diff
@@ -143,4 +143,49 @@
       vendor_files.must_equal vendor_cache_files
     end
   end
+
+  it 'vendors profiles when using a local path' do
+    local_depends_path = File.join(profile_path, 'local-depends')
+    dir_profile_path = File.join(profile_path, 'complete-profile')
+    tar_profile_path = File.join(profile_path,
+                         'archived-profiles',
+                         'tar_profile-1.0.0.tar.gz'
+                       )
+    zip_profile_path = File.join(profile_path,
+                         'archived-profiles',
+                         'zip_profile-1.0.0.zip'
+                       )
+
+    Dir.mktmpdir do |tmpdir|
+      [dir_profile_path, tar_profile_path, zip_profile_path].each do |profile|
+        included_tmpdir = File.join(tmpdir, File.basename(profile))
+        FileUtils.cp_r(profile, included_tmpdir)
+      end
+
+      profile_tmpdir = File.join(tmpdir, File.basename(local_depends_path))
+      FileUtils.cp_r(local_depends_path + '/.', profile_tmpdir)
+
+      out = inspec('vendor ' + profile_tmpdir + ' --overwrite')
+      out.stderr.must_equal ''
+      out.exit_status.must_equal 0
+
+      Dir.glob(File.join(profile_tmpdir, 'vendor', '*')).length.must_equal 3
+    end
+  end
+
+  it 'extracts archives in vendor directory when present' do
+    archive_depends_path = File.join(profile_path, 'archive-depends')
```

Comment:
This does not test #2041, which specifically mentioned that the `url:` fetcher worked, but `git`, `github`, and `compliance` do not.

Reply:
This is actually not tied to #2041 but to #3286 directly. The `github:` fetcher never existed and we have tests for ``url:` and `git:` in other places so I'm okay with the current functional coverage. The `compliance:` fetcher requires an Automate server so I don't think we can add tests for that.

All this being said, the reported issues all fall back to the previous behavior which would vendor archives as opposed to extracting them. This tests that any archive in `vendor` will be extracted.

Reply:
Removing #1974 and #2041 from the fix list here as this PR does not specifically prove those.

#2041 looks to be currently fixed already on master, I will do some additional testing and take care of this.

#1974 would be an extension and is not supported in this PR